xclim/core/calendar.py

```diff
@@ -491,10 +496,13 @@ def percentile_doy(
     rrr = rr.assign_coords(time=ind).unstack("time").stack(stack_dim=("year", "window"))
 
     if rrr.chunks is not None and len(rrr.chunks[rrr.get_axis_num("stack_dim")]) > 1:
-        # Preserve chunk size
-        time_chunks_count = len(arr.chunks[arr.get_axis_num("time")])
-        doy_chunk_size = np.ceil(len(rrr.dayofyear) / (window * time_chunks_count))
-        rrr = rrr.chunk(dict(stack_dim=-1, dayofyear=doy_chunk_size))
+        chunking = {"stack_dim": -1}
+        if keep_chunk_size:
```

Comment:
Not 100% sure it's useful, I was expecting better perfs when doing the chunking only once in bootstrap algo but it seems to not change anything.
I guess it's because map_block already computes things chunk by chunk and does not rechunk again each chunk